src/main/scala/org/apache/spark/sql/kinesis/HDFSMetadataCommitter.scala

```diff
@@ -165,4 +211,35 @@ class HDFSMetadataCommitter[T <: AnyRef : ClassTag](path: String,
     }
   }
 
+  /** Helper method to retry with exponential backoff  */
+  def withRetry[ T ](message: String, ignoreException: Boolean = true)(body: => T): T = {
+    var retryCount = 0
```

Comment:
retry with backoff seems pretty useful, should we move it to Utils so that it can be used in other places in spark code in future?

Reply:
Changes in spark code are outside the scope of this connector.